packages/core/src/repository/query/query.ts

```diff
@@ -32,25 +34,38 @@ export class AllObjectsQuery extends Query {
     }
 }
 
+/**
+ * @deprecated
+ */
 export class ObjectQuery<T> extends Query {
     constructor(public readonly value: T) {
         super();
     }
 }
 
+/**
+ * @deprecated
+ */
 export class ObjectsQuery<T> extends Query {
     constructor(public readonly values: T[]) {
         super();
     }
 }
 
+/**
+ * @deprecated
+ */
 export class ObjectRelationsQuery<T> extends ObjectQuery<T> {
     constructor(value: T, public readonly relations: string[] = []) {
         super(value);
     }
 }
 
-export abstract class PaginationQuery extends Query {}
+/**
+ * @deprecated
+ */
+export abstract class PaginationQuery extends Query {
```

Comment:
I think this slipped. We don't want to deprecate this, no?

Reply:
If it is deprecated, we should not touch the original code. Otherwise, current projects will be affected.

Reply:
Well, but to begin with, should we deprecate it?

Regarding `abstract` keyword, isn't it functionally the same with or without `abstract`? It doesn't have any member property/method.

Reply:
@doup  Good catch, fixed.